catalyst-gateway/bin/src/service/api/cardano/staking/assets_get.rs

```diff
@@ -136,145 +129,119 @@ async fn calculate_stake_info(
     session: &CassandraSession, stake_address: Cip19StakeAddress, slot_num: Option<SlotNo>,
 ) -> anyhow::Result<Option<StakeInfo>> {
     let address: StakeAddress = stake_address.try_into()?;
-    let mut txos_by_txn = get_txo_by_txn(session, &address, slot_num).await?;
-    if txos_by_txn.is_empty() {
+    let adjusted_slot_num = slot_num.unwrap_or(SlotNo::MAXIMUM);
+
+    let (mut txos, native_tokens) = futures::try_join!(
+        get_txo(session, &address, adjusted_slot_num),
+        get_native_tokens(session, &address, adjusted_slot_num)
+    )?;
+    if txos.is_empty() {
         return Ok(None);
     }
 
-    check_and_set_spent(session, &mut txos_by_txn).await?;
-    // TODO: This could be executed in the background, it does not actually matter if it
```

Comment:
why remove the comment and not fix the issue?
These should be two operations not one combined one, because s the comment says, setting spent should NEVER slow down the operation of this function, its an optimization.  This code forces it to be inline which is hugely suboptimal.  We never need to wait for the update to finish here, its not required to complete the function and even if it fails, nothing breaks.